great_expectations/datasource/data_connector/configured_asset_sql_data_connector.py

```diff
@@ -184,9 +184,12 @@ def get_batch_definition_list_from_batch_request(self, batch_request: BatchReque
         try:
             sub_cache = self._data_references_cache[batch_request.data_asset_name]
         except KeyError:
-            raise KeyError(
-                f"data_asset_name {batch_request.data_asset_name} is not recognized."
+            self.add_data_asset(
+                batch_request.data_asset_name,
+                {"table_name": batch_request.data_asset_name},
             )
+            self._refresh_data_references_cache()
+            sub_cache = self._data_references_cache[batch_request.data_asset_name]
```

Comment:
Could you explain this portion a bit? Just want to make sure I'm wrapping my head around it.

Reply:
If the `data_asset_name` provided in the notebook `batch_request` isn't found in the existing config, this will add that `data_asset` to the references available to the `validator` by assuming the `data_asset_name` provided refers to the name of a table in the sql datasource.

Reply:
Will this persist in the Data Context config, or is it just available in memory?

Reply:
@talagluck this persists in the Data Context config, and I have added a test confirming that.

Reply:
> @talagluck this persists in the Data Context config, and I have added a test confirming that.

This was not correct. This persists only in memory.

Reply:
@NathanFarmer @talagluck Is it proper to create a `data_asset` and put it into the configuration's memory where it has not existed originally, just because it was requested?  Thanks.

Reply:
@alexsherstinsky it is one of the requirements of this PR to allow the user to update the `data_asset` in the `suite new/edit` notebook. This cache is built from the data context config, but we want to allow users to pass `data_asset_name`s at runtime that do not exist in config if they are using a `ConfiguredAssetSqlDataConnector`. Removing this logic results in the `KeyError` because the `asset` doesn't exist in `great_expectations.yml`.

Reply:
@NathanFarmer ConfiguredAssetSqlDataConnector or InferredAssetSqlDataConnector for this case?

Reply:
@alexsherstinsky see the updated logic in `new_datasource.py` line 221:
- `ConfiguredAssetSqlDataConnector`
- `ConfiguredAssetFilePathDataConnector` (`ConfiguredAssetFilesystemDataConnector`)
- `RuntimeDataConnector`